pkg/module/drift/drift.go

```diff
@@ -291,31 +264,42 @@ func (d *Drift) handleIdentifier(identifiers []string) (exp.Expression, exp.Expr
 	}
 
 	if strings.Contains(identifiers[0], "${") {
-		return nil, nil, fmt.Errorf("identifier still contains variable")
+		return nil, fmt.Errorf("identifier still contains variable")
 	}
 
 	if usingVariable {
-		return goqu.L(identifiers[0]), goqu.L("(" + identifiers[0] + ") AS id"), nil
+		return goqu.L("(" + identifiers[0] + ") AS id"), nil
 	}
 
-	return goqu.I("c." + identifiers[0]), goqu.I("c." + identifiers[0]).As("id"), nil
+	return goqu.I("c." + identifiers[0]).As("id"), nil
 }
 
-func getIACProvider(provs []*cqproto.GetProviderSchemaResponse) (*cqproto.GetProviderSchemaResponse, error) {
-	var iacProv *cqproto.GetProviderSchemaResponse
-	for _, p := range provs {
-		if p.Name == string(iacTerraform) {
-			if iacProv != nil {
-				return nil, fmt.Errorf("only single IAC provider is supported at a time")
+func readIACStates(iacID string, stateFiles []string) (iacProvider, interface{}, error) {
+	if len(stateFiles) == 0 {
+		return "", nil, fmt.Errorf("state files for %s not specified", iacID)
+	}
+
+	switch iacProvider(iacID) {
+	case iacTerraform:
+		ret := make([]*terraform.Data, len(stateFiles))
+
+		fs := afero.NewOsFs()
+		for idx, fn := range stateFiles {
+			fh, err := fs.Open(fn)
+			if err != nil {
+				return "", nil, err
 			}
-			iacProv = p
+			data, err := terraform.ParseAndValidate(fh)
+			_ = fh.Close()
```

Comment:
multiple files would lead to resource problems as they won't get closed in order. would need to reindent with an inline fn or move it to another method. also I don't think ParseAndValidate can panic.